doc/plugin_server_upstreamauthority_vault.md

```diff
@@ -0,0 +1,89 @@
+#  Upstream Authority "vault" Plugin
+
+The vault plugin signs intermediate CA certificates for SPIRE using the Vault PKI Engine.
+The plugin does not support the `PublishJWTKey` RPC and is therefore not appropriate for use in nested SPIRE topologies where JWT-SVIDs are in use.
+
+## Configuration
+
+The plugin accepts the following configuration options:
+
+| key | type | required | description | default |
+|:----|:-----|:---------|:------------|:--------|
+| vault_host  | string |   | A Hostname of the Vault server or combination of Hostname and Port. | `${VAULT_HOST}` |
+| vault_path_prefix | string |  | A URL Path Prefix for the Vault API. e.g., If the Vault places behind the reverse proxy, it is possible to customize the URL Path.. e.g., the Vault places behind the reverse proxy, it is possible to customize the URL Path base. ( <vault_path_prefix>/v1/api/...) | `${VAULT_PATH_PREFIX}` |  
```

Comment:
My concern about `vault_addr` was that it required a URL but all that it seemed practical for people to change would be the host and port, since the scheme should be `https` and I didn't see a reason to specify a path.

If deploying behind behind a reverse proxy is a normal thing for people to do, where there might be a need to have a path prefix, and considering that the vault client APIs document the configurable as being a URL then maybe the original configurable was good? :grimacing:

Reply:
OK. I'll revert the commits.
In original case, Should we check the URI scheme ? 
e.g., If it's not https, outputs some warn logs.

Reply:
revert: 73782ba